specification/hybridcompute/resource-manager/Microsoft.HybridCompute/preview/2023-10-03-preview/privateLinkScopes.json

```diff
@@ -0,0 +1,1882 @@
+          "Reconciles the network security perimeter configuration of the private link scope.": {
+            "$ref": "./examples/networkSecurityPerimeterConfiguration/NetworkSecurityPerimeterConfigurationReconcile.json"
+          }
+        }
+      }
+    }
+  },
+  "definitions": {
+    "NetworkConfiguration": {
+      "type": "object",
+      "properties": {
+        "properties": {
+          "x-ms-client-flatten": true,
+          "$ref": "#/definitions/NetworkConfigurationProperties"
+        }
+      },
+      "allOf": [
+        {
+          "$ref": "../../../../../common-types/resource-management/v3/types.json#/definitions/ProxyResource"
+        }
+      ]
+    },
+    "NetworkConfigurationProperties": {
+      "type": "object",
+      "description": "Network configuration properties",
+      "properties": {
+        "location": {
+          "type": "string",
+          "description": "Resource location",
+          "x-ms-mutability": [
+            "create",
+            "read"
+          ]
+        },
+        "tenantId": {
+          "type": "string",
+          "readOnly": true,
+          "description": "Azure resource tenant Id"
+        },
+        "networkConfigurationScopeId": {
+          "type": "string",
+          "description": "Associated Network Configuration Scope Id"
+        },
+        "networkConfigurationScopeResourceId": {
+          "type": "string",
+          "description": "Associated Network Configuration Scope Resource Id"
+        },
+        "keyProperties": {
+          "description": "Public key information for client authentication",
+          "readOnly": true,
+          "$ref": "#/definitions/KeyProperties"
+        }
+      },
+      "required": [
+        "networkConfigurationScopeResourceId"
+      ]
+    },
+    "KeyProperties": {
+      "type": "object",
+      "description": "Public key information for client authentication",
+      "properties": {
+        "clientPublicKey": {
+          "description": "Current public key details",
+          "readOnly": true,
+          "$ref": "#/definitions/KeyDetails"
+        },
+        "candidatePublicKey": {
+          "description": "Candidate public key details",
+          "readOnly": true,
+          "$ref": "#/definitions/KeyDetails"
+        }
+      }
+    },
+    "KeyDetails": {
+      "type": "object",
+      "description": "Public key details",
+      "properties": {
+        "publicKey": {
+          "type": "string",
+          "description": "Public key",
+          "readOnly": true
+        },
+        "notAfter": {
+          "type": "string",
+          "format": "date-time",
+          "description": "Key expiration date",
+          "readOnly": true
+        },
+        "renewAfter": {
+          "type": "string",
+          "format": "date-time",
+          "description": "Recommended key renewal date",
+          "readOnly": true
+        }
+      }
+    },
+    "NetworkSecurityPerimeterConfiguration": {
```

Comment:
Consider extend this resource from https://github.com/Azure/azure-rest-api-specs/blob/main/specification/common-types/resource-management/v5/types.json#L9?